doc/README.md

```diff
@@ -81,3 +81,47 @@ Using caching, this means the first time you build the documentation, it might t
 After that, notebook execution is only triggered when you change the notebook source file.
 
 The benefits of working with notebooks for examples are that you don't separate the code from the documentation, but can still easily smoke-test the code.
+
+## Adding Markdown docs from external (ecosystem) repositories
+
+In order to avoid a situation where duplicate documentation files live in both the docs folder
+in this repository and in external repositories of ecosystem libraries (eg. xgboost-ray), you can
+specify Markdown files that will be downloaded from other GitHub repositories during the build process.
+
+In order to do that, simply edit the `EXTERNAL_MARKDOWN_FILES` list in `source/custom_directives.py`
+using the format in the comment. Before build process, the specified files will be downloaded, preprocessed
+and saved to given paths. The build process will then proceed as normal.
+
+Because GitHub Markdown and MyST differ in syntax, simple preprocessing is performed to allow for differences
+in rendering on GitHub and in docs. You can use two commands (`$UNCOMMENT` and `$REMOVE`/`$END_REMOVE`)
+in the Markdown file, specified in the following way:
+
+### `$UNCOMMENT`
+
+GitHub:
+
+```html
+<!--$UNCOMMENTthis will be uncommented--> More text
```

Comment:
I wonder if we can somehow make this convention easier in a follow-up PR. It'd be great to add more external content from trusted sources, without them having to add our special commenting syntax.